src/routing/core.rs

```diff
@@ -2557,7 +2594,7 @@ impl Core {
         // the actual last key of the resulting proof chain because it's the key that will be used
         // to sign the message.
         let additional_key = additional_key
-            .or_else(|| self.network.knowledge_by_name(&dst.name()?))
+            .or_else(|| self.network.key_by_name(&dst.name()?))
```

Comment:
this shall be our key that trusted by destination section.
not destination section's key .

Reply:
We don't now want this @maqi  is that right? Other sections key is not our concern, AE will update if it has to but we cannot know what they know? (anybody can update them)